docs/notebooks/tutorials/adv_newcv_subclass.ipynb

```diff
@@ -0,0 +1,509 @@
+    "- `ratio_losses` which set the relative weighting of the sketch-map/reconstruction loss.\n",
+    "- `high_dim_params` and `low_dim_params` which set the parameters of the sigmoid functions in the high/low dimensional spaces\n",
+    "\n",
+    "An instance of the SketchMapLoss is saved into `sketchmap_fn`, which allows us to modify the sigmoid default parameters after initialization.\n",
+    "\n",
+    "**training_step**\n",
+    "\n",
+    "Here we first compute a complete pass of the network (encoder+decoder) to evaluate the reconstruction loss. This is done by calling the method of the `super` class. \n",
+    "Then we evaluate the sketch map loss based on the inputs as well as the latent space and sum them. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mlcolvar.cvs import AutoEncoderCV\n",
+    "\n",
+    "class EncoderMap(AutoEncoderCV):\n",
+    "    \n",
+    "    def __init__(self, encoder_layers: list, decoder_layers: list = None, \n",
+    "                       ratio_losses : float = 100, \n",
+    "                       high_dim_params: dict = None,\n",
+    "                       low_dim_params: dict = None,\n",
+    "                       options: dict = None, **kwargs):\n",
+    "                       \n",
+    "        \"\"\"EncoderMap CV built from an AutoEncoderCV, optimized with both reconstruction and\n",
+    "        sketch-map losses.\n",
+    "\n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        ratio_losses : float, optional\n",
+    "            ratio of sketch-map vs MSE loss, by default 100\n",
+    "        high_dim_params : dict, optional\n",
+    "            dictionary of sigmoid parameters (input space)\n",
+    "        low_dim_params: dict, optional\n",
+    "            dictionary of sigmoid parameters (latent space)\n",
+    "\n",
+    "        See also\n",
+    "        --------\n",
+    "        mlcolvar.cvs.AutoEncoderCV\n",
+    "\n",
+    "        \"\"\"\n",
+    "        super().__init__(encoder_layers=encoder_layers, decoder_layers=decoder_layers, options=options, **kwargs)\n",
+    "\n",
+    "        # LOSS FUNCTIONS\n",
+    "        # Note: Autoencoder has MSE as primary loss, i.e., self.loss_fn = MSELoss()\n",
+    "        # here we add also the sketch-map loss (params can be set by changing its members)\n",
+    "        self.sketchmap_fn = SketchMapLoss()\n",
+    "\n",
+    "        # PARAMETERS\n",
+    "        self.ratio_losses = ratio_losses\n",
+    "\n",
+    "    def training_step(self, train_batch, batch_idx):\n",
+    "        \"\"\"Training step\"\"\"\n",
+    "        # 1) this computes reconstruction loss\n",
+    "        rec_loss = super().training_step(train_batch, batch_idx)\n",
+    "\n",
+    "        # 2) compute sketch-map loss\n",
+    "        X = train_batch['data']\n",
+    "        s = self.forward(X)\n",
```

Comment:
This should probably be ``forward_cv(x)``, and we might also explain the difference ``forward``/``forward_cv`` if we don't do it elsewhere.

Btw, the fact that partial results are hidden inside ``training_step()`` seems a common problem for inheritance. Because of this, we need to evaluate the ``encoder`` twice in each step here. In the future, we might think of a way to make this easier, like having the CVs implementing a ``evaluate_loss()`` method that takes a bunch of variables  that currently have scope only within ``training_step()`` (e.g., the result of the encoder).

Reply:
thanks for spotting the typo, I will correct it! 

I agree that we should discuss `forward`/`forward_cv` both in the tutorial related to implementing a cv from scratch (if not done yet) and in the pre/postprocessing tutorials.

regarding the second comment I will create an issue to remember discussing it in the future.